File: dashboard/src/models/audit.model.ts
export enum AuditKind {
    Catalog,
    Series,
    Provider
}

export enum AuditStatus {
    Queued,
    Processing,
    Completed,
    Failed
}

export interface AuditModel {
    job_id: number;
    kind: AuditKind;
    job_name: string;
    provider: string;
    status: AuditStatus;
    started_at: string;
    finished_at: string | null;
    error: string | null;
    expires: string | null;
}

File: server/AniStream/DTO/AuditModel.cs
using AniStream.Models;

namespace AniStream.API.DTO;

public enum AuditKind
{
    Catalog,
    Series,
    Provider
}

public sealed class AuditModel
{
    public required int JobId { get; set; }
    
    public required AuditKind Kind { get; set; }
    
    public required string JobName { get; set; }
    
    public required string Provider { get; set; }
    
    public required SyncJobStatus Status { get; set; }
    
    public required DateTime StartedAt { get; set; }
    
    public required DateTime? FinishedAt { get; set; }
    
    public required string? Error { get; set; }
    
    public required DateTime? Expires { get; set; }
}